backend/api/quivr_api/modules/rag_service/rag_service.py

```diff
@@ -144,29 +165,49 @@ async def generate_answer(
         logger.info(
             f"Creating question for chat {self.chat_id} with brain {self.brain.brain_id} "
         )
-        rag_config = await self._build_rag_config()
-        logger.debug(f"generate_answer with config : {rag_config.model_dump()}")
+        retrieval_config = await self._get_retrieval_config()
+        logger.debug(f"generate_answer with config : {retrieval_config.model_dump()}")
         history = await self.chat_service.get_chat_history(self.chat_id)
+        #  Format the history, sanitize the input
+        chat_history = self._build_chat_history(history)
+
         # Get list of files
-        list_files = await self.knowledge_service.get_all_knowledge_in_brain(
-            self.brain.brain_id
+        list_files = (
+            await self.knowledge_service.get_all_knowledge_in_brain(self.brain.brain_id)
+            if self.knowledge_service
+            else []
         )
+
         # Build RAG dependencies to inject
-        vector_store = self.create_vector_store(
-            self.brain.brain_id, rag_config.llm_config.max_input
+        vector_store = (
+            self.create_vector_store(
+                self.brain.brain_id, retrieval_config.llm_config.max_input_tokens
+            )
+            if self.vector_service
+            else None
         )
-        llm = self.get_llm(rag_config)
-        # Initialize the RAG pipline
-        rag_pipeline = QuivrQARAGLangGraph(
-            rag_config=rag_config, llm=llm, vector_store=vector_store
+
+        llm = self.get_llm(retrieval_config)
+
+        brain_core = BrainCore(
+            name=self.brain.name,
+            id=self.brain.id,
+            llm=llm,
+            vector_db=vector_store,
+            embedder=vector_store.embeddings if vector_store else None,
         )
-        #  Format the history, sanitize the input
-        chat_history = self._build_chat_history(history)
 
-        parsed_response = rag_pipeline.answer(question, chat_history, list_files)
+        parsed_response = brain_core.ask(
+            question=question,
+            retrieval_config=retrieval_config,
+            rag_pipeline=QuivrQARAGLangGraph,
+            list_files=list_files,
+            chat_history=chat_history,
+        )
 
         # Save the answer to db
-        new_chat_entry = self.save_answer(question, parsed_response)
+        if self.brain_service:
```

Comment:
Why do we need the brain_service to save chat answer ? This should depend on the `chat_service` ?